File: algorithm/insertion sort.py
def pra_insertion_sort(arr):
    for index in range(len(arr) - 1):
        for index2 in range(index + 1, 0, -1):
            if arr[index2] < arr[index2 - 1]:
                arr[index2], arr[index2 - 1] = arr[index2 - 1], arr[index2]
            else:
                break
    return arr


arr = [10, 2, 8, 1]
pra_insertion_sort(arr)

File: algorithm/insertion sort.js
const arr = [1, 20, 10, 2, 11];

const selectedSort = (arr) => {
  let i = 1,
    temp,
    j;
  for (i = 1; i < arr.length; i++) {
    temp = arr[i];

    for (j = i - 1; j >= 0 && temp < arr[j]; j--) {
      arr[j + 1] = arr[j];
    }
    arr[j + 1] = temp;
  }

  return arr;
};

selectedSort(arr);
